sw/host/opentitanlib/src/ownership/isfb.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+use serde::{Deserialize, Serialize};
+use serde_annotate::Annotate;
+use std::io::{Read, Write};
+
+use super::application_key::ApplicationKeyDomain;
+use super::misc::{TlvHeader, TlvTag};
+use super::GlobalFlags;
+
+/// The owner Integration Specific Firmware Binding (ISFB) configuration
+/// describes the configuration parameters for the ISFB region.
+#[derive(Debug, Serialize, Deserialize, Annotate)]
+pub struct OwnerIsfbConfig {
+    /// Header identifying this struct.
+    #[serde(
+        skip_serializing_if = "GlobalFlags::not_debug",
+        default = "OwnerIsfbConfig::default_header"
+    )]
+    pub header: TlvHeader,
+    /// The flash bank where the ISFB is located.
+    pub bank: u8,
+    /// The info flash page where the ISFB is located.
+    pub page: u8,
+    /// Padding for alignment.
+    #[serde(default, skip_serializing_if = "GlobalFlags::not_debug")]
+    #[annotate(format=hex)]
+    pub pad: u16,
+
+    /// The erase policy for the ISFB region.
+    #[annotate(format=hex)]
+    pub erase_conditions: u32,
+    /// The key domain associated with the erase condition policy.
+    #[annotate(format=hex)]
+    pub key_domain: ApplicationKeyDomain,
+    /// Reserved for future use.
+    #[serde(default, skip_serializing_if = "GlobalFlags::not_debug")]
+    #[annotate(format=hex)]
+    pub reserved: [u32; 5],
+    /// Number of `uint32_t` reserved for product expressions. It has to be a
+    /// value less than or equal to 256.
+    pub product_words: u32,
+}
+
+impl Default for OwnerIsfbConfig {
+    fn default() -> Self {
+        Self {
+            header: Self::default_header(),
+            bank: 0u8,
+            page: 0u8,
+            pad: 0u16,
+            erase_conditions: 0u32,
+            key_domain: ApplicationKeyDomain::default(),
+            reserved: [0u32; 5],
+            product_words: 0u32,
+        }
+    }
+}
+
+impl OwnerIsfbConfig {
+    const SIZE: usize = 2048;
```

Comment:
I don't think this struct is 2048 bytes.  I think its 26 words or 104 bytes.

There doesn't seem to be a length-dependency in rust for the length encoded in the TLV header (after all, this struct isn't variable sized), but this will definitely fail in the firmware if not correct.

Reply:
Fixed. This is by default 44 bytes.

The fix was introduced in https://github.com/lowRISC/opentitan/pull/25147.